azerothcore/azerothcore-wotlk
# feat(docker): production images, integrated ccache and many other improvements

## Changes Proposed:
- differentiate local vs production images and push them both on docker hub
- build cache between authserver and worldserver
- fix errors in build stage that also fixes the cache hit problem
- integrated ccache inside the dashboard compiler
- docker CI now caches the ccache
- Improved VSCode dev-container integration (added tasks)

## SOURCE:



## Tests Performed:


- Tested on Windows with WSL and Linux


## How to Test the Changes:

Follow the installation with docker: https://www.azerothcore.org/wiki/Install-with-Docker and check that everything works


## Known Issues and TODO List:



## Target Branch(es):







 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
